### PR TITLE
[generate_dump]: Enhance the generate_dump debugging script

### PR DESCRIPTION
**Pull request by justin-oliver**
**- What I did**
Added additional debugging information to the generate_dump script including
 - USB and PCI devices
 - Transceiver status and eeprom data
 - Interface status and counters
 - ASIC specific data
 - Arista specific info

**- How I did it**
Added additional commands to the generate_dump script. 
The ASIC version is determined using sonic_cfggen to parse the sonic_version YAML file. 
The Arista platform check is done by grepping /host/machine.conf as the Arista bootloader (Aboot) populates this file.

**- How to verify it**
admin@gd381:~$ show techsupport
...
/var/dump/sonic_dump_gd381_20180625_174535.tar.gz

admin@gd381:~$ tar -xzf /var/dump/sonic_dump_gd381_20180625_174720.tar.gz -C ~/

admin@gd381:~$ ls -l ~/sonic_dump_gd381_20180625_174720/dump
total 2620
-rw-r--r-- 1 admin admin  33533 Jun 25 17:47 APP_DB.json
-rw-r--r-- 1 admin admin 820099 Jun 25 17:47 ASIC_DB.json
-rw-r--r-- 1 admin admin 774247 Jun 25 17:47 COUNTERS_DB.json
**-rw-r--r-- 1 admin admin   6065 Jun 25 17:47 arista.dump
-rw-r--r-- 1 admin admin     69 Jun 25 17:47 arista.syseeprom**
-rw-r--r-- 1 admin admin      0 Jun 25 17:47 bgp.neighbors
-rw-r--r-- 1 admin admin      0 Jun 25 17:47 bgp.summary
-rw-r--r-- 1 admin admin     49 Jun 25 17:47 bgp.table.gz
**-rw-r--r-- 1 admin admin   6821 Jun 25 17:47 broadcom.ps
-rw-r--r-- 1 admin admin   1872 Jun 25 17:47 broadcom.soc
-rw-r--r-- 1 admin admin    781 Jun 25 17:47 broadcom.version**
-rw-r--r-- 1 admin admin    495 Jun 25 17:47 df
**-rw-r--r-- 1 admin admin 104858 Jun 25 17:47 dmesg
-rw-r--r-- 1 admin admin    992 Jun 25 17:47 docker.pmon
-rw-r--r-- 1 admin admin   1596 Jun 25 17:47 docker.ps**
-rw-r--r-- 1 admin admin    230 Jun 25 17:47 free
**-rw-r--r-- 1 admin admin   9588 Jun 25 17:47 interface.counters
-rw-r--r-- 1 admin admin   5168 Jun 25 17:47 interface.status
-rw-r--r-- 1 admin admin  20875 Jun 25 17:47 interface.xcvrs.eeprom
-rw-r--r-- 1 admin admin   1609 Jun 25 17:47 interface.xcvrs.presence**
-rw-r--r-- 1 admin admin  15122 Jun 25 17:47 ip.addr
-rw-r--r-- 1 admin admin  11883 Jun 25 17:47 ip.link
-rw-r--r-- 1 admin admin    270 Jun 25 17:47 ip.neigh
-rw-r--r-- 1 admin admin   8397 Jun 25 17:47 ip.route
-rw-r--r-- 1 admin admin    275 Jun 25 17:47 ip.rule
-rw-r--r-- 1 admin admin   6042 Jun 25 17:47 lldpctl
**-rw-r--r-- 1 admin admin  99323 Jun 25 17:47 lspci
-rw-r--r-- 1 admin admin  16524 Jun 25 17:47 lsusb
-rw-r--r-- 1 admin admin    193 Jun 25 17:47 machine.conf**
-rw-r--r-- 1 admin admin   2091 Jun 25 17:47 mount
**-rw-r--r-- 1 admin admin     54 Jun 25 17:47 platform.psustatus**
-rw-r--r-- 1 admin admin     77 Jun 25 17:47 platform.summary
-rw-r--r-- 1 admin admin     86 Jun 25 17:47 platform.syseeprom
-rw-r--r-- 1 admin admin  17560 Jun 25 17:47 ps.aux
-rw-r--r-- 1 admin admin 263865 Jun 25 17:47 saidump
**-rw-r--r-- 1 admin admin   1174 Jun 25 17:47 scd**
-rw-r--r-- 1 admin admin   2517 Jun 25 17:47 sensors
-rw-r--r-- 1 admin admin 316214 Jun 25 17:47 sysctl
-rw-r--r-- 1 admin admin   1810 Jun 25 17:47 version
-rw-r--r-- 1 admin admin    573 Jun 25 17:47 vmstat
-rw-r--r-- 1 admin admin  10765 Jun 25 17:47 vmstat.m
-rw-r--r-- 1 admin admin    764 Jun 25 17:47 vmstat.s

**- Previous command output (if the output of a command-line utility has changed)**
N/A

**- New command output (if the output of a command-line utility has changed)**
N/A

### CODE REVIEW

File: scripts/generate_dump [hunk 1]
```diff
@@ -294,6 +308,19 @@ main() {
...
     fi
 
+    local asic="$(/usr/local/bin/sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)"
+    if [ "$asic" = "broadcom" ]; then
+        save_cmd "bcmcmd -t5 version" "broadcom.version"
+        save_cmd "bcmcmd -t5 soc" "broadcom.soc"
+        save_cmd "bcmcmd -t5 ps" "broadcom.ps"
+    fi
+
+    if $GREP -qi aboot /host/machine.conf; then
```
Comment (qiluo-msft) [inline scripts/generate_dump]: >$GREP -qi aboot /host/machine.conf [](start = 7, length = 34)

Check $aboot_platform?

Comment (justin-oliver) [reply scripts/generate_dump]: Are you suggesting I source /host/machine.conf and then check the $aboot_platform environment variable?

Comment (qiluo-msft) [reply scripts/generate_dump]: Either way:
1. source /host/machine.conf, and check $aboot_platform.
2. grep "aboot_platform="

---
In reply to: [197968999](https://github.com/Azure/sonic-utilities/pull/272#discussion_r197968999) [](ancestors = 197968999)

Comment (justin-oliver) [reply scripts/generate_dump]: I implemented option 2. The Arista platform check is now more explicit.

**Pull request merged**